class.jetpack-post-images.php

```diff
@@ -806,6 +806,66 @@ public static function get_attachment_data( $attachment_id, $post_url = '', $wid
 		);
 	}
 
+	/**
+	 * Get info about a WordPress attachment in a Story block.
+	 *
+	 * @since 9.1.0
+	 *
+	 * @param int    $attachment_id Attachment ID.
+	 * @param string $post_url      URL of the post, if we have one.
+	 * @param int    $width         Minimum Image width.
+	 * @param int    $height        Minimum Image height.
+	 * @return array|bool           Image data or false if unavailable.
+	 */
+	public static function get_attachment_data_for_story( $attachment_id, $post_url = '', $width, $height ) {
+		if ( empty( $attachment_id ) ) {
+			return false;
+		}
+
+		$meta = wp_get_attachment_metadata( $attachment_id );
+
+		if ( empty( $meta ) ) {
+			return false;
+		}
+
+		if ( ! empty( $meta['videopress'] ) ) {
+			// Use poster image for VideoPress videos.
+			$url         = $meta['videopress']['poster'];
+			$meta_width  = $meta['videopress']['width'];
+			$meta_height = $meta['videopress']['height'];
+		} elseif ( ! empty( $meta['thumb'] ) ) {
```

Comment:
Could you clarify with an inline comment what scenario this is meant to catch?

Reply:
Done in b2011c2, let me know if you have anything better in mind for the comment 👍